js/app/warehouse.js

```diff
@@ -12,16 +12,17 @@ var Warehouse = new Knowledge.Class({
     _init: function (props={}) {
         this.parent(props);
         this._local_modules = new Map();
+        this._importer = imports.app.modules;
+        this._custom_importer = imports['.'];
+        this._custom_importer.searchPath = [];
+        this._custom_importer.searchPath.unshift('resource:///app/js');
     },
 
     register_class: function (module_name, module_class) {
         this._local_modules.set(module_name, module_class);
     },
 
-    type_to_class: function (module_name) {
-        if (this._local_modules.has(module_name))
-            return this._local_modules.get(module_name);
-
+    _load_module: function (module_name, importer) {
```

Comment:
As long as you're changing these lines, use the method shorthand syntax `_load_module(module_name, importer) {`

Reply:
Ill change the other methods as well, I prefer that the same file is consistent in these things.